shotover-proxy/src/config/topology.rs

```diff
@@ -95,22 +93,27 @@ impl Topology {
 
 #[cfg(test)]
 mod topology_tests {
+    use crate::config::chain::TransformChainConfig;
     use crate::config::topology::Topology;
     use crate::transforms::coalesce::CoalesceConfig;
+    use crate::transforms::debug::printer::DebugPrinterConfig;
+    use crate::transforms::null::NullSinkConfig;
+    use crate::transforms::TransformConfig;
     use crate::{
         sources::{redis::RedisConfig, Sources, SourcesConfig},
         transforms::{
             distributed::tuneable_consistency_scatter::TuneableConsistencyScatterConfig,
             parallel_map::ParallelMapConfig, redis::cache::RedisConfig as RedisCacheConfig,
-            TransformsConfig,
         },
     };
     use std::collections::HashMap;
     use tokio::sync::watch;
 
-    async fn run_test_topology(chain: Vec<TransformsConfig>) -> anyhow::Result<Vec<Sources>> {
+    async fn run_test_topology(
```

Comment:
Could this function just box the vec's elements itself. Save a bunch of boilerplate `Box::new()`

Reply:
It cant because we need to erase the full type in order to store different types within the same vec, and its the Box that does the type erasure